playwright/src/main/java/com/microsoft/playwright/impl/UrlMatcher.java

```diff
@@ -77,21 +78,32 @@ static String resolveUrl(URL baseUrl, String spec) {
   }
 
   private UrlMatcher(URL baseURL, String glob, Pattern pattern, Predicate<String> predicate) {
-    this.baseURL = baseURL;
+    this.baseURL = baseURL != null ? baseURL.toString() : null;
     this.glob = glob;
     this.pattern = pattern;
     this.predicate = predicate;
   }
 
   boolean test(String value) {
+    return testImpl(baseURL, pattern, predicate, glob, value);
+  }
+
+  private static boolean testImpl(String baseURL, Pattern pattern, Predicate<String> predicate, String glob, String value) {
     if (pattern != null) {
       return pattern.matcher(value).find();
     }
     if (predicate != null) {
       return predicate.test(value);
     }
     if (glob != null) {
-      return Pattern.compile(globToRegex(resolveUrl(baseURL, glob))).matcher(value).find();
+      if (!glob.startsWith("*")) {
+        // Allow http(s) baseURL to match ws(s) urls.
```

Comment:
This mimics https://github.com/microsoft/playwright/blob/640e6a8aa75a23a31e7591f34176a32ed73ba022/packages/playwright-core/src/utils/isomorphic/urlMatch.ts#L101-L103